### PR TITLE
Error handling cleanup in solvers/prop

### PR DESCRIPTION
**Pull request by danpoe**

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: bd5d901a35146a1cac8d91343171e948c2268288).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/84743220

**Review by peterschrammel — APPROVED** [review]

File: src/solvers/prop/prop_conv.cpp [hunk 1]
```diff
@@ -35,22 +35,17 @@ void prop_convt::set_frozen(const bvt &bv)
 
 bool prop_conv_solvert::literal(const exprt &expr, literalt &dest) const
 {
-  assert(expr.type().id()==ID_bool);
+  PRECONDITION(expr.type().id() == ID_bool);
+  PRECONDITION(expr.id() == ID_symbol);
```
Comment (tautschnig) [inline src/solvers/prop/prop_conv.cpp]: Hmm, this actually seemed to invite a slightly more extensive change: #2989.

File: src/solvers/prop/prop_conv.cpp [hunk 2]
```diff
@@ -221,22 +219,21 @@ literalt prop_conv_solvert::convert_bool(const exprt &expr)
...
   }
   else if(expr.id()==ID_if)
   {
-    if(op.size()!=3)
-      throw "if takes three operands";
-
-    return prop.lselect(convert(op[0]), convert(op[1]), convert(op[2]));
+    const if_exprt &if_expr = to_if_expr(expr);
+    return prop.lselect(
+      convert(if_expr.op0()), convert(if_expr.op1()), convert(if_expr.op2()));
```
Comment (tautschnig) [inline src/solvers/prop/prop_conv.cpp]: Use `.cond`, `.true_case`, `.false_case`

**Review by tautschnig — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: b98ed58c8bb3be275f1e5e907901f61cf3c46863).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85587985

**Pull request merged**